sled-agent/src/rack_setup/service.rs

```diff
@@ -734,16 +734,16 @@ impl ServiceInner {
 
         let rack_network_config = {
             let config = &config.rack_network_config;
-            NexusTypes::RackNetworkConfigV1 {
+            NexusTypes::RackNetworkConfigV2 {
                 rack_subnet: config.rack_subnet,
                 infra_ip_first: config.infra_ip_first,
                 infra_ip_last: config.infra_ip_last,
                 ports: config
                     .ports
                     .iter()
-                    .map(|config| NexusTypes::PortConfigV1 {
```

Comment:
For lack of a better place to put this comment: You should also change the `schema_version` to `2` in the `EarlyNetworkConfig` on line 1139. As we discussed, the schema version doesn't work as I expected or really matter for correctness right now, due to our use of json and not having an intermediates serialized body. But for consistency, it should still be changed I think.